pkg/services/correlations/database.go

```diff
@@ -99,32 +99,42 @@ func (s CorrelationsService) updateCorrelation(ctx context.Context, cmd UpdateCo
 			return ErrSourceDataSourceReadOnly
 		}
 
-		if cmd.Label == nil && cmd.Description == nil {
+		if cmd.Label == nil && cmd.Description == nil && cmd.Config == nil {
```

Comment:
nit: currently if we send an update request with
```json
{
  "config": {}
}
```
the api returns a 200 without actually updating anything, where it should maybe return a 400

we could solve this by simply doing:
```suggestion
		if cmd.Label == nil && cmd.Description == nil && (cmd.Config == nil || (cmd.Config.Field == nil && cmd.Config.Target == nil && cmd.Config.Type == nil)) {
```
or implement a `Validate` function like we did for `CorrelationConfigType`, but it's not actually causing any issue and it's pretty much an edge case, we can revisit this later if needed.